src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -331,14 +331,19 @@ public void fireFileStarted(String fileName) {
     @Override
     public void fireErrors(String fileName, SortedSet<LocalizedMessage> errors) {
         final String stripped = CommonUtils.relativizeAndNormalizePath(basedir, fileName);
+        boolean removeCache = false;
         for (final LocalizedMessage element : errors) {
             final AuditEvent event = new AuditEvent(this, stripped, element);
             if (filters.accept(event)) {
+                removeCache = true;
                 for (final AuditListener listener : listeners) {
                     listener.addError(event);
                 }
             }
         }
+        if (removeCache && cache != null) {
+            cache.remove(fileName);
+        }
```

Comment:
@romani This can't be done without an API change to `MessageDispatcher` or a temporary field variable.
`cache.put` requires the timestamp of the file and the method signature doesn't receive that variable.
Method signature only gives us fileName and it seems like a hack to use that to re-retrieve the file's timestamp.